packages/sdks/src/blocks/image.lite.tsx

```diff
@@ -16,9 +16,92 @@ export interface ImageProps {
   children?: any;
   fitContent?: boolean;
   builderBlock?: any;
+  noWebp?: boolean;
 }
 
 export default function Image(props: ImageProps) {
+  function updateQueryParam(
+    uri = '',
+    key: string,
+    value: string | number | boolean
+  ): string {
+    const re = new RegExp('([?&])' + key + '=.*?(&|$)', 'i');
+    const separator = uri.indexOf('?') !== -1 ? '&' : '?';
+    if (uri.match(re)) {
+      return uri.replace(
+        re,
+        '$1' + key + '=' + encodeURIComponent(value) + '$2'
+      );
+    }
+
+    return uri + separator + key + '=' + encodeURIComponent(value);
+  }
+
+  function removeProtocol(path: string) {
+    return path.replace(/http(s)?:/, '');
+  }
+
+  function getShopifyImageUrl(src: string, size: string): string | null {
+    if (!src || !src?.match(/cdn\.shopify\.com/) || !size) {
+      return src;
+    }
+
+    if (size === 'master') {
+      return removeProtocol(src);
+    }
+
+    const match = src.match(
+      /(_\d+x(\d+)?)?(\.(jpg|jpeg|gif|png|bmp|bitmap|tiff|tif)(\?v=\d+)?)/i
+    );
+
+    if (match) {
+      const prefix = src.split(match[0]);
+      const suffix = match[3];
+      const useSize = size.match('x') ? size : `${size}x`;
+
+      return removeProtocol(`${prefix[0]}_${useSize}${suffix}`);
+    }
+
+    return null;
+  }
+
+  function getSrcSet(url: string): string {
+    if (!url) {
+      return url;
+    }
+
+    const sizes = [100, 200, 400, 800, 1200, 1600, 2000];
+
+    if (url.match(/builder\.io/)) {
+      let srcUrl = url;
+      const widthInSrc = Number(url.split('?width=')[1]);
+      if (!isNaN(widthInSrc)) {
+        srcUrl = `${srcUrl} ${widthInSrc}w`;
+      }
+
+      return sizes
+        .filter((size) => size !== widthInSrc)
+        .map((size) => `${updateQueryParam(url, 'width', size)} ${size}w`)
+        .concat([srcUrl])
+        .join(', ');
+    }
+
+    if (url.match(/cdn\.shopify\.com/)) {
+      return sizes
+        .map((size) => [getShopifyImageUrl(url, `${size}x${size}`), size])
+        .filter(([sizeUrl]) => !!sizeUrl)
+        .map(([sizeUrl, size]) => `${sizeUrl} ${size}w`)
+        .concat([url])
+        .join(', ');
+    }
+
+    return url;
+  }
+
+  function useSrcSet() {
```

Comment:
to solve this we just need to update the jsx parser to merge the two things (e.g. grab state object and merge in any additional state hooks or top level functions) instead of use just one or the other. which should be very easy